packages/plugin-hardhat/src/prepare-upgrade.ts

```diff
@@ -23,8 +23,12 @@ export type PrepareUpgradeFunction = (
   opts?: PrepareUpgradeOptions,
 ) => Promise<DeployImplementationResponse>;
 
-export function makePrepareUpgrade(hre: HardhatRuntimeEnvironment): PrepareUpgradeFunction {
+export function makePrepareUpgrade(hre: HardhatRuntimeEnvironment, platformModule: boolean): PrepareUpgradeFunction {
   return async function prepareUpgrade(proxyOrBeacon, ImplFactory, opts: PrepareUpgradeOptions = {}) {
+    if (platformModule && opts.platform === undefined) {
+      opts.platform = true;
+    }
```

Comment:
Similar comment about mutation here. Plus let's have an `enablePlatform` helper?

Reply:
The helper is named as `withPlatformDefaults`